src/modal.ts

```diff
@@ -0,0 +1,462 @@
+            let tmp = parseAttrVal(new_val);
+            this._guide_value = {}
+            for (let k of Object.keys(tmp))
+                this._guide_value[tmp[k]] = k;
+        }
+    }
+    constructor() {
+        super();
+        document.addEventListener("keydown", this.handleKey.bind(this));
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+    }
+    override disconnectedCallback() {
+        this.clearTimer();
+    }
+    handleKey(e: KeyboardEvent) {
+        if(!this.noesc && this.open && e.key === 'Escape')
+            if (!active_modals?.length ||
+                active_modals[active_modals.length - 1] == this)
+                this.hide();
+    }
+    clearTimer() {
+        if (this.timer_obj)
+            clearTimeout(this.timer_obj);
+    }
+    containerClicked() {
+        if(this.open && !this.nobgclose)
+            this.hide();
+    }
+    _wrapperClickedEvent(event:any) {
+        event.stopPropagation();
+        if (this.closesel && event.target.matches(this.closesel))
+            this.hide();
+    }
+    show(delay?:number) {
+        this.clearTimer();
+        if (this._open)
+            return;
+        if(delay)
+            this.timer_obj = setTimeout( () => {this.timer_obj=null;
+                                        this.show()}, delay);
+        else {
+            if(this._wrp_init_bounds) {
+               this._wrp_init_bounds = '';
+               this._wrapper_el.style.right = '';
+               this._wrapper_el.style.left = '';
+               this._wrapper_el.style.width = '';
+               this._wrapper_el.style.top = '';
+               this._wrapper_el.style.bottom = '';
+               this._wrapper_el.style.height = '';
+            }
+            this.style.zIndex = DEFAULT_MODAL_ZINDEX;
+            if(this.stack)
+                this.style.zIndex = DEFAULT_MODAL_ZINDEX + active_modals.length;
+            else
+                active_modals.forEach((m:any) => m.hide());
+            if(!this.actualParent)
+                this.actualParent = this.parentElement;
+            this._open = this.open = true;
+            active_modals.push(this);
+            document.body.appendChild(this);
+        }
+    }
+    hide(delay?:number) {
+        this.clearTimer();
+        if (!this._open)
+            return;
+        if (delay)
+            this.timer_obj = setTimeout( () => {this.timer_obj=null;
+                                        this.hide()}, delay);
+        else
+            this.open = false;
+        if (this.animationDur == 0)
+            this.closeEnd();
+
+    }
+    closeEnd() {
+        this._open = this.open;
+        if (!this._open) {
+            this.actualParent?.appendChild(this);
+            let ix = active_modals.indexOf(this);
+            if (ix >= 0)
+                active_modals.splice(ix, 1);
+        }
+    }
+    private _dragStart(event:any) {
+        if (event.target.matches(Object.keys(this._guide_value)))
```

Comment:
`matches()` accepts string parameter.  Instead of sending an array you can convert to a comma separated string by doing `Object.keys(this._guide_value).join(",")`